common/rpc.go

```diff
@@ -65,3 +69,31 @@ type (
 		GetMaxMessageSize() int
 	}
 )
+
+// GetClientHeaders returns the headers that should be sent from client to server
+func GetClientHeaders(ctx context.Context) map[string]string {
+	call := yarpc.CallFromContext(ctx)
+	headerNames := call.HeaderNames()
+	headerExists := map[string]struct{}{}
+	for _, h := range headerNames {
+		headerExists[h] = struct{}{}
```

Comment:
I don't feel strongly about this, but I recall Steven making a point about headers not... technically being a map, since they support duplicates under some implementations.